IanGRap/Squish
# Treasure Grab Scoring Refactor

Contributor: Coulter

Goal: Rewrite how per-round scoring works to make it more like how scoring is done in Block Dunk.

Important Files:
- TreasureGrabModeManager.cs: Added an array to store player scores, updated over time. Now updating player scores during play no longer requires the Manager to communicate with ActivePlayers.cs.
- TreasureUI.cs: Updated to only require the parameters of the manager event in order to update score text, rather than needing to talk to ActivePlayers.cs.
- ActivePlayers.cs: Removed all functions related to "Score".
- PlayerData.cs: Removed Score field of PlayerData class.
- MatchManager.cs: This script is obsolete, so it has been deleted.

How to test:
- Start up the Treasure Grab scene.
- Play through the game, keeping an eye on player scores. Make sure they are updated correctly when players break chests and/or die. You'll know the changes basically worked if everything behaves the same as it used to.
